src/shared/type-coercions.js

```diff
@@ -55,15 +114,61 @@ export function toObject(value) {
 }
 
 /**
- * Returns the value as a string or an empty string if the
- * value undefined, null or otherwise falsely.
+ * Returns either an object or null.
  *
- * @param {any} value - initial value
- * @return {string}
+ * @type {(value: any) => Object|null}
  */
-export function toString(value) {
-  if (value && typeof value.toString === 'function') {
-    return value.toString();
+toObjectShim.orNull = value => (value === null ? null : toObjectShim(value));
+
+/**
+ * Returns the value from the custom `shape` formatter.  This is different
+ * than `toCustom` in that this ensures the value is an object type before
+ * the custom shape formatter is run.
+ *
+ * Usage:
+ *
+ *  toObject.shape(value => {
+ *    return {
+ *       foo: coercions.toString(value.foo),
+ *       bar: coercions.toInteger(value.bar),
+ *    };
+ *  })
+ *
+ * @param {(value: any) => Object} shape - Custom coercion formatter.
+ */
+toObjectShim.shape = shape => {
+  /**
+   * Custom object shim.
+   *
+   * @param {any} value - Initial value
+   */
+  function toObjectShapeShim(value) {
+    return shape(toObjectShim(value));
   }
-  return '';
-}
+  /**
+   * Returns either the custom `shape` formatter value or null.
+   *
+   * Usage:
+   *
+   *  toObject.shape(value => {
+   *    ...
+   *  }).orNull
+   *
+   * @type {(value: any) => Object|null}
+   */
+  toObjectShapeShim.orNull = value =>
+    value === null ? null : toObjectShapeShim(value);
+  return toObjectShapeShim;
+};
+
+// Exported coercion types
+const coercions = {
+  toString: toStringShim,
+  toInteger: toIntegerShim,
+  toObject: toObjectShim,
+  toBoolean: toBooleanShim,
+  custom: customShim,
```

Comment:
toObject.shape and custom are similar, the former will ensure the object is in fact an object before the custom shape is called, whereas custom take the raw value of any time. I felt both were different enough to warrant both. Currently, toObject.shape is not used anywhere, but I suspect we will use it if we extend this.